dev: add failing test for deployment of contract account when there is no return data

### Context

When we deploy contract accounts, it is currently failing for bytecode that has no return data such as:
- 0x
- 0x6000600155600160015500

This results in this error

```Python
starkware.cairo.lang.vm.vm_exceptions.VmException: /Users/fredtupas/kkrt-labs/kakarot/src/kakarot/instructions/system_operations.cairo:589:13: Error at pc=0:16714:
E           Unknown value for memory cell at address 181:3618502788666131213697322783095070105623107215331596699973092056135872020480.
E                       [ctx.sub_context.return_data - 1],
...
E           /Users/fredtupas/kkrt-labs/kakarot/src/kakarot/instructions.cairo:648:31: (pc=0:18243)
E                               let ctx = CallHelper.finalize_calling_context(ctx);
E                                         ^**************************************^
```

--- a/tests/integration/solidity_contracts/PlainOpcodes/test_plain_opcodes.py
+++ b/tests/integration/solidity_contracts/PlainOpcodes/test_plain_opcodes.py
@@ -192,14 +192,17 @@ class TestPlainOpcodes:
             """
             TODO: need to fix when there is no return data from the bytecode execution,
             it calls CallHelper instead of CreateHelper when finalizing calling context
+            https://github.com/kkrt-labs/kakarot/issues/726
             """
         )
         @pytest.mark.parametrize("bytecode", ["0x", "0x6000600155600160015500"])
-        async def test_should_create_contract_with_bytecode_and_no_return_data(
+        async def test_should_create_empty_contract_when_creation_code_has_no_return(
             self,
             plain_opcodes,
             plain_opcodes_deployer,
             bytecode,
+            get_starknet_address,
+            get_contract_account,
         ):
             evm_addresses = await plain_opcodes.create(
                 bytecode=bytecode,
@@ -207,17 +210,13 @@ class TestPlainOpcodes:
                 caller_address=plain_opcodes_deployer,
             )
 
-            # Assert that the contract was deployed with the correct bytecode.
+            # Assert that the contract was deployed with no bytecode
             assert len(evm_addresses) == 1
-            actual_bytecode = "".join(
-                [
-                    f"{x:02x}"
-                    for x in (
-                        await plain_opcodes.contract_account.bytecode().call()
-                    ).result.bytecode
-                ]
+            contract_account = get_contract_account(
+                get_starknet_address(evm_addresses[0])
             )
-            assert bytecode[2:] == actual_bytecode
+            actual_bytecode = (await contract_account.bytecode().call()).result.bytecode
+            assert actual_bytecode == []
 
     class TestCreate2:
         async def test_should_deploy_bytecode_at_address(

--- a/tests/integration/solidity_contracts/PlainOpcodes/test_plain_opcodes.py
+++ b/tests/integration/solidity_contracts/PlainOpcodes/test_plain_opcodes.py
@@ -189,7 +189,10 @@ class TestPlainOpcodes:
             assert nonce_final == nonce_initial + count
 
         @pytest.mark.skip(
-            "TODO: need to fix when there is no return data from the bytecode execution in EVMInstructions.run"
+            """
+            TODO: need to fix when there is no return data from the bytecode execution,
+            it calls CallHelper instead of CreateHelper when finalizing calling context
+            """
         )
         @pytest.mark.parametrize("bytecode", ["0x", "0x6000600155600160015500"])
         async def test_should_create_contract_with_bytecode_and_no_return_data(
@@ -198,21 +201,21 @@ class TestPlainOpcodes:
             plain_opcodes_deployer,
             bytecode,
         ):
-            nonce_initial = (
-                await plain_opcodes.contract_account.get_nonce().call()
-            ).result.nonce
-
             evm_addresses = await plain_opcodes.create(
                 bytecode=bytecode,
                 count=1,
                 caller_address=plain_opcodes_deployer,
             )
-            assert len(evm_addresses) == 1
 
-            nonce_final = (
-                await plain_opcodes.contract_account.get_nonce().call()
-            ).result.nonce
-            assert nonce_final == nonce_initial + 1
+            # Assert that the contract was deployed with the correct bytecode.
+            assert len(evm_addresses) == 1
+            assert bytecode[2:] == "".join(
+                str(
+                    (
+                        await plain_opcodes.contract_account.bytecode().call()
+                    ).result.bytecode
+                )
+            )
 
     class TestCreate2:
         async def test_should_deploy_bytecode_at_address(

--- a/tests/integration/solidity_contracts/PlainOpcodes/test_plain_opcodes.py
+++ b/tests/integration/solidity_contracts/PlainOpcodes/test_plain_opcodes.py
@@ -213,7 +213,7 @@ class TestPlainOpcodes:
             # Assert that the contract was deployed with no bytecode
             assert len(evm_addresses) == 1
             contract_account = get_contract_account(
-                get_starknet_address(int(evm_addresses[0]))
+                get_starknet_address(int(evm_addresses[0], 16))
             )
             actual_bytecode = (await contract_account.bytecode().call()).result.bytecode
             assert actual_bytecode == []

--- a/tests/integration/solidity_contracts/PlainOpcodes/test_plain_opcodes.py
+++ b/tests/integration/solidity_contracts/PlainOpcodes/test_plain_opcodes.py
@@ -209,13 +209,15 @@ class TestPlainOpcodes:
 
             # Assert that the contract was deployed with the correct bytecode.
             assert len(evm_addresses) == 1
-            assert bytecode[2:] == "".join(
-                str(
-                    (
+            actual_bytecode = "".join(
+                [
+                    f"{x:02x}"
+                    for x in (
                         await plain_opcodes.contract_account.bytecode().call()
                     ).result.bytecode
-                )
+                ]
             )
+            assert bytecode[2:] == actual_bytecode
 
     class TestCreate2:
         async def test_should_deploy_bytecode_at_address(

--- a/tests/integration/solidity_contracts/PlainOpcodes/test_plain_opcodes.py
+++ b/tests/integration/solidity_contracts/PlainOpcodes/test_plain_opcodes.py
@@ -188,6 +188,32 @@ class TestPlainOpcodes:
             ).result.nonce
             assert nonce_final == nonce_initial + count
 
+        @pytest.mark.skip(
+            "TODO: need to fix when there is no return data from the bytecode execution in EVMInstructions.run"
+        )
+        @pytest.mark.parametrize("bytecode", ["0x", "0x6000600155600160015500"])
+        async def test_should_create_contract_with_bytecode_and_no_return_data(
+            self,
+            plain_opcodes,
+            plain_opcodes_deployer,
+            bytecode,
+        ):
+            nonce_initial = (
+                await plain_opcodes.contract_account.get_nonce().call()
+            ).result.nonce
+
+            evm_addresses = await plain_opcodes.create(
+                bytecode=bytecode,
+                count=1,
+                caller_address=plain_opcodes_deployer,
+            )
+            assert len(evm_addresses) == 1
+
+            nonce_final = (
+                await plain_opcodes.contract_account.get_nonce().call()
+            ).result.nonce
+            assert nonce_final == nonce_initial + 1
+
     class TestCreate2:
         async def test_should_deploy_bytecode_at_address(
             self,

--- a/tests/integration/solidity_contracts/PlainOpcodes/test_plain_opcodes.py
+++ b/tests/integration/solidity_contracts/PlainOpcodes/test_plain_opcodes.py
@@ -213,7 +213,7 @@ class TestPlainOpcodes:
             # Assert that the contract was deployed with no bytecode
             assert len(evm_addresses) == 1
             contract_account = get_contract_account(
-                get_starknet_address(evm_addresses[0])
+                get_starknet_address(int(evm_addresses[0]))
             )
             actual_bytecode = (await contract_account.bytecode().call()).result.bytecode
             assert actual_bytecode == []

